Update Link UI design for preview action buttons to use icons for edit and unlink

getdave opened the pull request:
<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/repository-management.md#pull-requests. -->

<!-- Gutenberg's license is in the process of updating to be dual-licensed under the GPL and MPL. As part of that transition, all new contributions are dual-licensed. For more information, see: https://github.com/WordPress/gutenberg/blob/trunk/LICENSE.md -->

<img width="708" alt="Screen Shot 2021-10-21 at 13 01 44" src="https://user-images.githubusercontent.com/444434/138273101-27f73447-e568-4d52-a379-0483f41015d7.png">


## Description
In https://github.com/WordPress/gutenberg/pull/33849#issuecomment-930368677 @jasmussen proposed a new design for the Link UI's preview mode. 

This PR implements part of that design by normalising the placement of the edit and unlink buttons.

Note that the work to add the split text/url fields is under way in as separate PR https://github.com/WordPress/gutenberg/pull/33849.

## How has this been tested?
1. Create link.
2. Click on link to bring up link preview.
3. See edit and unlink action buttons are now adjacent to the link title and in icon form.

## Screenshots <!-- if applicable -->

### Before 
<img width="642" alt="Screen Shot 2021-10-21 at 13 03 32 (1)" src="https://user-images.githubusercontent.com/444434/138273416-910f2892-167e-4de6-b4bc-8256cee8e601.png">

### After

<img width="708" alt="Screen Shot 2021-10-21 at 13 01 44" src="https://user-images.githubusercontent.com/444434/138273101-27f73447-e568-4d52-a379-0483f41015d7.png">

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [ ] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [ ] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [ ] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->

getdave committed:
Add unlink and edit buttons in icon form

jasmussen commented:
Nice one, here's a GIF of the two buttons in action:
![nice](https://user-images.githubusercontent.com/1204802/138281846-c864940e-1a15-41f0-99bd-8d184731bf02.gif)

Looks good, works well. 

Technically unrelated to this PR, so feel free to ignore, but the link card descriptio has `black` text, it should be `$gray-900`:
<img width="398" alt="Screenshot 2021-10-21 at 14 56 51" src="https://user-images.githubusercontent.com/1204802/138282165-31e285c9-1f2f-4fc4-ad2c-ba493bf71f18.png">

It looks like that color is coming directly from the components themselves, and it seems like those components should not be setting colors to anything but inherit. @ciampo would this be an easy fix?

Also not directly related, but the focus style of the link is both cropped and the oldschool wp-admin link style:

<img width="311" alt="Screenshot 2021-10-21 at 14 57 01" src="https://user-images.githubusercontent.com/1204802/138282539-aa152d66-4d00-4494-8d6a-8fc875debf6a.png">

This is probably also best fixed or ticketed separately. It might even look different in the site editor already, since that's iframed.

Nice work! What do you think? I'd be happy to give a green check on this one as is.

getdave commented:
Thanks for review @jasmussen.

> Nice work! What do you think? I'd be happy to give a green check on this one as is.

Happy to address both the items you noticed as follow ups. Should be simple enough.

I also need to fix/update the tests.

jasmussen commented:
Mostly to clarify, it doesn't necessarily have to be your responsibility to catch all of those. But for the purposes of the review, I think it's worth calling the various glitches out explicitly, as it might not be obvious that the glitch is a separate issue.

getdave commented:
> the link card descriptio has black text, it should be $gray-900:

Addressed in https://github.com/WordPress/gutenberg/pull/35851

> but the focus style of the link is both cropped and the oldschool wp-admin link style:

Addressed in https://github.com/WordPress/gutenberg/pull/35853

>  It might even look different in the site editor already, since that's iframed.

Nope it looks the same for whatever reason (see screenshots above).

> Mostly to clarify, it doesn't necessarily have to be your responsibility to catch all of those. But for the purposes of the review, I think it's worth calling the various glitches out explicitly, as it might not be obvious that the glitch is a separate issue.

I didn't fully understand this. I think you're saying I should call out things like this before I request PR review. I agree, but unfortunately my poor design eyes didn't spot these details 😄 More than happy to address them though.

getdave committed:
Use a11y means to selecting Edit button

getdave committed:
Fix e2e tests by targetting aria label

jasmussen commented:
> I didn't fully understand this. I think you're saying I should call out things like this before I request PR review. I agree, but unfortunately my poor design eyes didn't spot these details 😄 More than happy to address them though.

Not at all, you're doing excellent work and I'm trying to clarify you don't have to fix everything under the sun, we're all here to help carry the load. 

What I'm also saying is that I have a habit of looking at the full flow when testing these PRs, and in the process I will often encounter small issues that are outside of the scope of a particular PR to fix. It isn't always obvious why a bug that is visible in a PR review is _not_ fixed as part of that same PR, so explaining why, I find, can be helpful. For example in the focus style PR you just did, it became very clear that updating from the blurry gray-blue box shadow to the new crisp theme-color solid style, that is something to fix in wp-admin itself, not Gutenberg.

getdave commented:
Thanks @jasmussen 

> Looks good, works well.

Do we have a 👍 on this one in that case?

jasmussen reviewed (approved):
Dig it:
<img width="884" alt="Screenshot 2021-10-25 at 11 00 19" src="https://user-images.githubusercontent.com/1204802/138666697-40124b6a-7c7b-40ad-8427-3265573d2e92.png">

As a note to future iterations, we can probably polish the URL preview card even more, to make a few more things line up more nicely here:

<img width="411" alt="Screenshot 2021-10-25 at 11 00 44" src="https://user-images.githubusercontent.com/1204802/138666815-ed06178e-e976-46ef-8744-578132a60abb.png">

But that's for another time.

getdave commented:
Thank you! 🙇 

Rebasing now (in hope of fixing seemingly unrelated randomly failing e2e test) with a view to merge.

getdave committed:
Force test run. Will revert.

getdave committed:
Revert "Force test run. Will revert."

This reverts commit 3d847218c1d70767d8b824c454f3cea02eff9600.
